src/webapp/gradebookServer.js

```diff
@@ -280,6 +280,57 @@ app.get('/sections', function(request, response) {
    });
 });
 
+
+//Returns an array of dates for a given section id.  Omits dates when classes are
+// not held.
+app.get('/sectionschedule', function(request, response) {
+   //Decrypt the password recieved from the client.  This is a temporary development
+   //feature, since we don't have ssl set up yet
+   var passwordText = sjcl.decrypt(superSecret, JSON.parse(request.query.password));
+
+   //Connnection parameters for the Postgres client recieved in the request
+   var config = createConnectionParams(request.query.user, request.query.database,
+      passwordText, request.query.host, request.query.port);
+
+   var sectionID = request.query.sectionID;
+
+   var queryText = 'SELECT date FROM significantDate WHERE classesheld = false;';
```

Comment:
```suggestion
   var queryText = 'SELECT date::VARCHAR FROM significantDate WHERE NOT classesheld;';
```

Casting `date` to a `VARCHAR` keeps it as `YYYY-MM-DD` when parsing. Otherwise, the `pg` module (or possibly something else) changes the date format to `YYYY-MM-DD1HH:MM:SS.222Z` (1 is a `T` character, 2s represent milliseconds). This is more difficult to parse, and falsely suggests having a significant time component.

The proposed alternative way of checking the boolean is clearer IMO.

Reply:
I did not realize the date format would be changed.  Good catch.